src/lib/Syntax.hs

```diff
@@ -216,8 +235,9 @@ data UPat' = UPatBinder UBinder
            | UPatCon Name (Nest UPat)
            | UPatPair UPat UPat
            | UPatUnit
-           | UPatRecord (LabeledItems UPat)
-           | UPatVariant Label Int UPat
+           | UPatRecord (ExtLabeledItems UPat UPat)
+           | UPatVariant (LabeledItems ()) Label UPat
+           | UPatVariantLift (LabeledItems ()) UPat
```

Comment:
Can you please put some comments at those expressions? I'm not sure what piece of syntax does `UPatVariantLift` represent? Is it for the `{| a | <expr> |}` pattern?

Reply:
That's right. Will add.